arris_tg2492lg/connect_box.py

```diff
@@ -30,16 +31,15 @@ class ConnectBox:
     def __init__(self, websession: ClientSession, hostname: str, password: str):
         self._websession = websession
         self._hostname = hostname
-        self._password = password
+        self._password = quote(password)
         self._nonce = str(random.randrange(10000, 100000))
         self._credential: Optional[Credential] = None
 
     async def async_login(self) -> str:
         arg_string = f"{USERNAME}:{self._password}"

```

Comment:
`arg_string = f"{quote(USERNAME)}:{quote(self._password)}"`

Although `USERNAME` (admin) doesn't need the be escaped I prefer to keep this in line with how the router portal works. If the username would ever be configurable I won't forget to escape it :)